test/gtest/fold.cpp

```diff
@@ -0,0 +1,160 @@
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright (c) 2024 Advanced Micro Devices, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+
+#include "fold.hpp"
+#include "miopen/bfloat16.hpp"
+#include "tensor_holder.hpp"
+#include <miopen/env.hpp>
+
+MIOPEN_DECLARE_ENV_VAR_BOOL(MIOPEN_TEST_ALL)
+
+namespace fold {
+struct FoldForwardTestFloat32 : FoldFwdTest<float>
+{
+};
+
+struct FoldForwardTestFloat16 : FoldFwdTest<half>
+{
+};
+
+struct FoldForwardTestBFloat16 : FoldFwdTest<bfloat16>
+{
+};
+
+struct FoldBackwardTestFloat32 : FoldBwdTest<float>
+{
+};
+
+struct FoldBackwardTestFloat16 : FoldBwdTest<half>
+{
+};
+
+struct FoldBackwardTestBFloat16 : FoldBwdTest<bfloat16>
+{
+};
+}; // namespace fold
+using namespace fold;
+TEST_P(FoldForwardTestFloat32, FoldForwardTest)
+{
+    if(miopen::IsEnabled(ENV(MIOPEN_TEST_ALL)))
```

Comment:
Please update the test invokation condition like this: https://github.com/moreh-dev/MIOpen/pull/27#discussion_r1634984024

Reply:
resolve in [e76095e](https://github.com/moreh-dev/MIOpen/pull/37/commits/e76095e8fc1771a82978af13018b538b396877d1)